NAS2D/Filesystem.cpp

```diff
@@ -291,34 +292,35 @@ void Filesystem::del(const std::string& filename)
 
 std::string Filesystem::readFile(const std::string& filename) const
 {
-	PHYSFS_file* myFile = PHYSFS_openRead(filename.c_str());
-	if (!myFile)
+	const auto& filePath = findFirstPath(filename, mSearchPaths);
+	if (filePath.empty())
 	{
-		closeFile(myFile);
-		throw std::runtime_error("Error opening file for reading: " + filename + " : " + getLastPhysfsError());
+		throw std::runtime_error("Error opening file: " + filename + " : File does not exist");
 	}
 
-	// Ensure that the file size is greater than zero and can fit in a std::string::size_type
-	auto fileLength = PHYSFS_fileLength(myFile);
-	if (fileLength < 0 || static_cast<PHYSFS_uint64>(fileLength) > std::numeric_limits<std::string::size_type>::max())
+	std::ifstream file{filePath, std::ios::in | std::ios::binary};
+	if (!file)
 	{
-		closeFile(myFile);
-		throw std::runtime_error("Error determining length of file or file too large: " + filename + " : Length = " + std::to_string(fileLength));
+		throw std::runtime_error("Error opening file: " + filename + " : " + errorDescription());
+	}
+
+	const auto fileSize = std::filesystem::file_size(filePath);
+	if constexpr (std::numeric_limits<decltype(fileSize)>::max() > std::numeric_limits<std::string::size_type>::max())
```

Comment:
Small nitpick, but it's important for cross-platform compatibility. `<windows.h>` defines its own `max` and `min` macros. In order to get around this you must surround all uses of the `numeric_limits` versions with parenthesis in order for the parser to not freak out:

`if constexpr ((std::numeric_limits<decltype(fileSize)>::max)() > (std::numeric_limits<std::string::size_time>::max)())`

Reply:
I'm not sure what you mean. The Windows CI builds didn't show any warnings or errors. When exactly is this a problem?

Also, Filesystem.cpp doesn't include `<windows.h>`.

Reply:
>
> Edit: I assume you mean, had we included `<windows.h>`, it would have been a problem.

Yes, if `<windows.h>` is ever included *anywhere* (third-party libraries as well). Even with `NO_MINMAX` defined, the problem presents itself.

Reply:
I don't think that's an overly large concern in this case. The code is in the implementation file, not the header file, so the include environment is fairly well controlled, and there is really no reason to include `<windows.h>` in `Filesystem.cpp`.